superset-frontend/src/reports/actions/reportState.js

```diff
@@ -112,3 +112,39 @@ export function toggleActive(report, isActive) {
       });
   };
 }
+
+export function deleteActiveReport(report) {
+  return function deleteActiveReportThunk(dispatch, getState) {
+    return SupersetClient.delete({
+      endpoint: encodeURI(`/api/v1/report/${report.id}`),
+    })
+      .then(() => {
+        const state = getState();
+        const { user, dashboardInfo, charts, explore } = state;
+        if (dashboardInfo) {
```

Comment:
did you want to move these into finally so that they don't show the report could not be deleted error?

Reply:
just added